src/components/radio/radio.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { createRef } from 'react'
+import { fireEvent, render, screen } from '@testing-library/react'
+import { Radio } from './radio'
+
+test('renders radio label correctly', () => {
+  render(<Radio label="Radio" />)
+  expect(screen.getByText('Radio')).toBeInTheDocument()
+})
+
+test('selection for radio working correctly', () => {
+  const onChange = jest.fn((event) => event.target.checked)
+  render(<Radio label="Radio" onChange={onChange} />)
+  fireEvent.click(screen.getByText('Radio'))
+  expect(onChange).toBeCalled()
+  expect(onChange.mock.results[0].value).toBe(true)
+})
+
+test('disable property of radio working correctly', () => {
```

Comment:
Also add a test to make sure that disabled radio buttons, don't call `onChange` on click.